### PR TITLE
Track Export: Sync tags for exported tracks before copying

### PR DESCRIPTION
**Pull request by ywwg**
Note that I re-enabled id3 sync to make this easier to test, the final merge will revert that change.

Testing: I did an export, the tags synced just fine

### CODE REVIEW

File: src/preferences/dialog/dlgpreflibrary.cpp [hunk 1]
```diff
@@ -39,7 +39,7 @@ DlgPrefLibrary::DlgPrefLibrary(QWidget * parent,
           m_iOriginalTrackTableRowHeight(Library::kDefaultRowHeightPx) {
     setupUi(this);
     slotUpdate();
-    checkbox_ID3_sync->setVisible(false);
+    checkbox_ID3_sync->setVisible(true);
```
Comment (ywwg) [inline src/preferences/dialog/dlgpreflibrary.cpp]: will be reverted before merge

File: src/preferences/dialog/dlgpreflibrarydlg.ui [hunk 2]
```diff
@@ -188,9 +188,6 @@
       </item>
       <item row="1" column="0" colspan="2">
        <widget class="QCheckBox" name="checkbox_ID3_sync">
-        <property name="enabled">
```
Comment (ywwg) [inline src/preferences/dialog/dlgpreflibrarydlg.ui]: will be reverted before merge

**Comment by daschuer** [comment]
I do not understand what this pr for. The Id sync feature schould care to write the tags to the original file independent from the export feature.
But I can think of a use case where the user wants to leave the original files untouched but have the Mixxx tags integrated into the exported files. This would require an extra preference option or export dialog option. In any case, the tags schould be written to the exported files. Note: taglib uses in place metadata edit.

**Comment by daschuer** [comment]
.. Or writing metadata could be the default, during export without any preference option.

**Comment by ywwg** [comment]
> .. Or writing metadata could be the default, during export without any preference option.

I was thinking the same thing, but this would update the metadata of the original file before the copy occurs.  If the user doesn't want their id3 tags synced, then this would be unexpected behavior

**Comment by ywwg** [comment]
> The Id sync feature schould care to write the tags to the original file independent from the export feature.

Unfortunately if the user enables or disables the sync feature occasionally, some tracks may be up to date and others may not be.  Doing the sync here ensures that the files are definitely up to date.

Alternatively, when turning on the sync preference, we could bulk sync all tracks in the library to ensure they are all correct, but that could take a very long time.

**Comment by daschuer** [comment]
We are here in the export feature, IMHO this should not at all depend on the metadata sync feature of the whole library. This feature here, is not responsible to fix the "metadata" out of sync issue. 
I would prefer only to write to the source tracks if sync is enabled and the metadata are touched explicite by the track preferences, or by a mass sync feature, similar to the "reload track metadata feature.

**Comment by rryan** [comment]
> This feature here, is not responsible to fix the "metadata" out of sync issue.

It would be nice if we could say "sync any tracks to be exported that haven't already been synced" though -- no? In which case this is an incomplete version of that feature since we can't currently keep track of which files have been dirtied since we last synced the metadata back to the file.

**Comment by uklotzde** [comment]
Related work:
- Implicit and explicit export of track metadata into files (one-way sync): #728 
- Implicit import of track metadata from files (experimental two-way sync with known drawbacks, but suits my personal use case): https://github.com/uklotzde/mixxx/tree/ReloadAudioTags
- Ideas for reliable synchronization of metadata by tracking the time stamp of the last import/export action: https://github.com/mixxxdj/mixxx/blob/master/src/track/trackrecord.h

**Comment by Be-ing** [comment]
What is the status of this PR?

**Comment by Holzhaus** [comment]
@uklotzde Considering all your metadata sync changes since 2016, is this PR still relevant? If not, let's close this.

**Comment by uklotzde** [comment]
I am not going to review draft PRs based on a version that is outdated since years. The comments might not apply any longer. Let's close it.

**Pull request closed without merging**